### DISCUSSION TITLE
자료구조 (가은 -> 진)

### DISCUSSION THREAD

**Discussion by gaeunnlee** [tempdku/temp#16] [General]
@jjinleee 

1. 큐와 우선순위 큐의 차이는 무엇인가요?

2. 자가 균형 이진 탐색 트리에 대하여 설명해주세요.

**Comment by jjinleee** [comment]
1.  큐는 FIFO 원칙을 따르는 자료구조로 삽입,삭제 연산을 사용합니다. 우선순위 큐는 큐와 달리 각 요소가 우선순위를 가지는 자료구조로 가장 높은 우선순위 먼저 처리되는 식으로 연산을 수행합니다. 우선순위큐는 작업 스케줄링, 다익스트라 알고리즘에  쓰이며 큐는 BFS, 프린터 대기열에 쓰입니다.
2. 자가 균형 이진 탐색 트리는 연산을 수행하면서 높이가 일정 범위내로 유지하도록 단일회전, 이중회전 등을 통해 균형을 유지한다. 대표적인 종류로 AVL tree, 레드 블랙 트리가 있습니다. 이러한 트리는 데이터 구조와 알고리즘의 효율성을 높이기 위해 사용합니다. 삽입, 삭제, 검색 연산이 항상 O(log N) 의 시간복잡도를 가집니다.
